firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -226,93 +358,77 @@ public void onClick(DialogInterface dialogInterface, int i) {
     return alertDialog;
   }
 
-  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
-  @NonNull
-  public Task<Void> signInTester() {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInCancellationSource.cancel();
-    }
-
-    signInCancellationSource = new CancellationTokenSource();
-    signInTaskCompletionSource = new TaskCompletionSource<>(signInCancellationSource.getToken());
-
-    Context context = firebaseApp.getApplicationContext();
-    AlertDialog alertDialog = getSignInAlertDialog(context);
-    alertDialog.show();
-
-    return signInTaskCompletionSource.getTask();
-  }
-
-  private void setSignInTaskCompletionError(FirebaseAppDistributionException e) {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInTaskCompletionSource.setException(e);
-    }
+  private OnSuccessListener<String> getFidGenerationOnSuccessListener(Context context) {
+    return new OnSuccessListener<String>() {
+      @Override
+      public void onSuccess(String fid) {
+        Uri uri =
+            Uri.parse(
+                String.format(
+                    SIGNIN_REDIRECT_URL,
+                    firebaseApp.getOptions().getApplicationId(),
+                    fid,
+                    getApplicationName(context),
+                    context.getPackageName()));
+        openSignInFlowInBrowser(uri);
+      }
+    };
   }
 
-  /** Returns true if the App Distribution tester is signed in */
-  @NonNull
-  public boolean isTesterSignedIn() {
-    return false;
-  }
+  AppDistributionRelease getLatestReleaseFromClient(
+      String fid, String appId, String apiKey, String authToken) {
+    AppDistributionRelease latestRelease = null;
+    try {
+      AppDistributionRelease retrievedLatestRelease =
+          firebaseAppDistributionTesterApiClient.fetchLatestRelease(fid, appId, apiKey, authToken);
 
-  /** Signs out the App Distribution tester */
-  public void signOutTester() {}
+      Context context = firebaseApp.getApplicationContext();
+      long currentInstalledVersionCode = getInstalledAppVersionCode(context);
```

Comment:
Inline the `context` here.  The reasoning is that it's more succinect, but also by inlining you show that it's not used elsewhere in this scope